lib/stages/BeamAssemble.hpp

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2022 Kotekan Developers
+
+/****************************************************
+* @file   BeamAssemble.hpp
+* @brief  This file declares the BeamAssemble class which
+*         takes an input buffer with frames for single 
+*         frequency, and assembles the frames for a given 
+*         number of frequencies in a larger buffer.
+*
+* @author Mehdi Najafi
+* @date   28 AUG 2022
+*****************************************************/
+
+#ifndef KOTEKAN_BEAMASSEMBLE_HPP
+#define KOTEKAN_BEAMASSEMBLE_HPP
+
+#include "Config.hpp"           // for Config
+#include "Stage.hpp"            // for Stage
+#include "bufferContainer.hpp"  // for bufferContainer
+
+/**
+ * @class BeamAssemble
+ * @brief Assembles incoming beam information from the frames in 
+ *        the in_buf and gather in a larger format to an out_buf.
+ * *
+ * @par Buffers
+ * @buffer in_buf The buffer to receive data from.
+ *     @buffer_format 4+4-bit complex voltage (beam) data
+ *     @buffer_metadata BeamMetadata
+ * @buffer out_buf The buffer to put the data to.
+ *     @buffer_format 4+4-bit complex voltage (beam) data
+ *     @buffer_metadata BeamMetadata
+ * 
+ * @par num_freq_per_output_frame
+ *      number of frequencies to be assembled
+ * 
+ * @par arriving_data_timeout
+ *      timeout in miliseconds for the arriving data
+ * 
+ * @par beam_printout
+ *      check flag to print some information about incoming beam 
+ *      data
+ * 
+ * @par late_beam_printout
+ *      check flag to print some information about late incoming 
+ *      beam data, which are ignored
+ * 
+ * @par received_beam_frames_count
+ *      a counter to keep track of the number of beam frames received
+ * 
+ * @par accepted_beam_frames_count
+ *      a counter to keep track of the number of beam frames received
+ *      and copied to the output buffer frames
+ *
+ * @par missed_beam_frames_count
+ *      a counter to keep track of the number of beam frames received
+ *      late and ignored (not copied to any output buffer frames)
```

Comment:
The config options should use the `@conf` custom doxygen tag.  The idea being we should be able to generate nice autodocs off these headers once we get time to work on that.

Reply:
Fixed.